[nur-haziq] iP

nur-haziq opened the pull request.

nur-haziq committed:
Add greet and exit

nur-haziq committed:
Rename chatbot

Chatbot renamed from "Duke" to "Mona".
ASCII Art updated.

nur-haziq committed:
Add greet and exit methods

The 'greet' and 'exit' output prints are now wrapped in a method
called greet() and exit() respectively.

nur-haziq committed:
Add echo

Bot now echoes commands entered by user

nur-haziq committed:
Implement 'add' and 'list' methods

Methods implemented:
*addTask() - Stores text entered by user
*printList() - Displays all text stored by user

nur-haziq committed:
Add Task.java

Implemented a class to represent each task.

nur-haziq committed:
Mona.java: Add ability to mark and unmark tasks

Tasks are stored in an array of Task objects, with the ability to
mark and unmark each task.

nur-haziq committed:
Task.java: Remove blank lines

nur-haziq committed:
Mona.java: Rename 'listofTasks' to 'tasks'

To be in line with Java coding standard.

nur-haziq committed:
Add Todo.java

Add skeleton code for Todo class

nur-haziq committed:
Add Deadline.java

Add skeleton code for Deadline class

nur-haziq committed:
Add Event.java

Add skeleton code for Event class

nur-haziq committed:
feat: Add ability for user to add 3 different types of tasks

User can now add:
- ToDos
- Deadlines
- Events

nur-haziq committed:
input.txt: Add input test cases

nur-haziq committed:
EXPECTED.txt: Add expected output for input test cases

JingHaoooo commented on src/main/java/Mona.java:
```diff
@@ -0,0 +1,106 @@
+        printHorizontalLine();
+    }
+    public static void main(String[] args) {
+        String logo = "___  ___                  \n"
+                + "|  \\/  |                  \n"
+                + "| .  . | ___  _ __   __ _ \n"
+                + "| |\\/| |/ _ \\| '_ \\ / _` |\n"
+                + "| |  | | (_) | | | | (_| |\n"
+                + "\\_|  |_/\\___/|_| |_|\\__,_|\n";
```
Any reason why you didn't make logo a constant?

JingHaoooo commented on src/main/java/Mona.java:
```diff
@@ -0,0 +1,106 @@
+        Task[] tasks = new Task[100];        //initialize an array of Tasks, to act as a list
+
+        Scanner in = new Scanner(System.in);
+        String line = in.nextLine();         // read in 1st command from user
```
Comment is not needed as the line is self explanatory.

JingHaoooo commented on src/main/java/Mona.java:
```diff
@@ -0,0 +1,106 @@
+            case ("bye"):
+                exit();
+                return;
+            default:
+                if (line.startsWith("mark")) {
+                    int i = Integer.parseInt(line.substring(5));
+                    tasks[i - 1].markAsDone();
+                    printMarkStatement(i - 1, tasks);
+                }
+                else if (line.startsWith("unmark")) {
+                    int i = Integer.parseInt(line.substring(7));
+                    tasks[i - 1].markAsNotDone();
+                    printUnmarkStatement(i - 1, tasks);
+                } else {
+                    addTask(line, tasks); // if command cannot be extracted, add text to list
```
Perhaps you can consider creating a new method for mark and unmark to avoid potential deep nesting.

JingHaoooo commented on src/main/java/Mona.java:
```diff
@@ -0,0 +1,106 @@
+
+        greet();
+
+        Task[] tasks = new Task[100];        //initialize an array of Tasks, to act as a list
```
Can consider making a constant MAX_TASK to store the maximum number of task.

annnniexu commented on src/main/java/Mona.java:
```diff
@@ -0,0 +1,106 @@
+        int index = 1;
+        for (Task task: tasks) {
+            if (task != null) {
+                System.out.println(Integer.toString(index) + "."
```
good job breaking up a long line

annnniexu commented on src/main/java/Mona.java:
```diff
@@ -0,0 +1,106 @@
+        printHorizontalLine();
+    }
+    public static void main(String[] args) {
+        String logo = "___  ___                  \n"
```
this is cool

annnniexu commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,19 @@
+public class Task {
+    protected String description;
```
naming is good

JingHaoooo reviewed (commented):
Overall, the code follows the code quality guidelines.

annnniexu commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,19 @@
+    public void markAsNotDone() {
+        isDone = false;
+    }
+}
```
did not see any coding standard violations.

nur-haziq committed:
Add ConsolePrint.java

Added skeleton code for ConsolePrint class. Class will be used to
print default responses to the console.

nur-haziq committed:
Add InputParser.java

InputParser class will be used to process user inputs.

nur-haziq committed:
Added TaskManager.java

TaskManager class will be used to manage tasks and execute commands to
update the list of stored tasks.

nur-haziq committed:
Mona.java: Refactor code to utilize InputParser and TaskManager classes

nur-haziq committed:
EXPECTED.txt: Update expected test output

Add "Now you have [x] tasks in the list." line as part of the
expected output when tasks are added to the list.

nur-haziq committed:
Task.java: Remove unused "noOfTasks" variable

nur-haziq committed:
Add Constants.java

Add a public class to store constants.

nur-haziq committed:
Refactor code to utilize Constants class

nur-haziq committed:
Add MonaException.java

Add custom exception class to handle user input errors

nur-haziq committed:
Add InputValidator.java

Add new class, InputValidator, consisting of methods that check for
valid user input

nur-haziq committed:
TaskManager.java: Change 'noOfTasks' variable to public static

nur-haziq committed:
InputParser.java: Refactor code to utilize InputValidator methods

InputParser is now able to deal with errors resulting from incorrect
user input

nur-haziq committed:
Constants.java: Add command types as constants

Commands such as "todo", "mark" etc. are added as constants.

nur-haziq committed:
Merge branch 'branch-Level-5'

nur-haziq committed:
Organize classes into multiple packages

Packages are organized in the following manner:
- exception: MonaException.java
- input: InputParser.java, InputValidator.java
- manager: TaskManager.java
- output: ConsolePrint.java
- task: Deadline.java, Event.java, Todo.java, Task.java
- util: Constants.java

nur-haziq committed:
Merge branch 'branch-A-Packages'

nur-haziq committed:
TaskManager.java: Refactor code to use ArrayList<Task> to store tasks

nur-haziq committed:
ConsolePrint.java: Update argument of printList method

Method now takes in an ArrayList<Task> object, instead of a Task array

nur-haziq committed:
Update test files

input.txt test script now contains incorrect input

nur-haziq committed:
feat: Add ability for user to delete tasks

nur-haziq committed:
Add error handling for incorrect 'delete' inputs

nur-haziq committed:
Add TaskStorage.java

Add skeleton code for TaskStorage class, a class that will be used
to handle saving and loading of data from a stored file

nur-haziq committed:
Add method to get description from Task subclasses

nur-haziq committed:
feat: Add ability to load and store data from the hard disk

Tasks are now saved in the hard disk, and loaded up when the chatbot
starts up.

nur-haziq committed:
Merge branch 'branch-Level-6'

nur-haziq committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/mona/manager/TaskManager.java

nur-haziq committed:
TaskStorage.java: Update argument of generateTextToAdd method

nur-haziq committed:
TaskManager.java: Refactor loadFromStorage method

nur-haziq committed:
Add MANIFEST.MF

To create JAR file

nur-haziq committed:
Refactor code to include loadData() method in TaskStorage.java

TaskStorage class will now handle loading of data from a .txt file,
instead of TaskManager

nur-haziq committed:
Add Ui.java

Add skeleton code for Ui class, a class which will be used to
handle inputs from the user.

nur-haziq committed:
Rename TaskStorage.java to Storage.java

nur-haziq committed:
Mona.java: Refactor code to use Ui class to handle user inputs

nur-haziq committed:
InputParser.java: Refactor code to handle invalid  user commands

nur-haziq committed:
InputValidator.java: Rename methods as verbs

nur-haziq committed:
Rename classes

- TaskManager renamed to TaskList
- InputParser renamed to Parser

nur-haziq committed:
Parser.java: Add method to extract details from 'find' command

nur-haziq committed:
InputValidator.java: Add method to check for proper 'find' input

nur-haziq committed:
feat: Add ability to find a task by searching for a keyword

nur-haziq committed:
ConsolePrint.java: Add method to print a filtered list

nur-haziq committed:
Deadline.java: Add ability to store data as a LocalDateTime object

nur-haziq committed:
InputValidator.java: Add JavaDoc comments

nur-haziq committed:
Ui.java: Add JavaDoc comments

nur-haziq committed:
Parser.java: Add JavaDoc comments

nur-haziq committed:
TaskList.java: Add JavaDoc comments

nur-haziq committed:
Storage.java: Add JavaDoc comments

nur-haziq committed:
Task.java: Add JavaDoc comments

nur-haziq committed:
Deadline.java: Add JavaDoc comments

nur-haziq committed:
Event.java: Add JavaDoc comments

nur-haziq committed:
Todo.java: Add JavaDoc comments

nur-haziq committed:
Mona.java: Add JavaDoc comments

nur-haziq committed:
Merge pull request #1 from nur-haziq/branch-Level-8

Deadline.java: Add ability to store data as a LocalDateTime object

nur-haziq committed:
Merge branch 'master' into branch-Level-9

nur-haziq committed:
Merge branch 'master' into branch-A-JavaDoc

# Conflicts:
#	src/main/java/mona/task/Deadline.java

nur-haziq committed:
Merge pull request #2 from nur-haziq/branch-Level-9

Add ability to find a task by searching for a keyword

nur-haziq committed:
Merge branch 'master' into branch-A-JavaDoc

# Conflicts:
#	src/main/java/mona/input/Parser.java

nur-haziq committed:
Merge pull request #3 from nur-haziq/branch-A-JavaDoc

Add JavaDoc comments to codebase

okkhoy commented on src/main/java/Mona.java:
```diff
@@ -0,0 +1,54 @@
+        this.ui = new Ui();
+        this.storage = new Storage(Constants.DATA_FILE_PATH);
+        this.taskList = new TaskList((this.storage).loadData());
+    }
+    /**
```
need some vertical spacing between the 2 methods to improve readability.
pls fix all instances.

okkhoy commented on src/main/java/mona/input/InputValidator.java:
```diff
@@ -0,0 +1,87 @@
+     */
+    public void checkMarkUnmarkDeleteCommand(String[] commandTypeAndParams) throws
+            MonaException, NumberFormatException {
+        // Checking if description (number) field is empty
```
redundant comment

okkhoy commented on src/main/java/mona/input/Parser.java:
```diff
@@ -0,0 +1,217 @@
+     *
+     * @param line The input line from which the command type is to be extracted.
+     */
+    public void extractCommandTypeFromString(String line) {
```
👍

okkhoy commented on src/main/java/mona/input/Parser.java:
```diff
@@ -0,0 +1,217 @@
+
+        switch(commandType) {
+        case ("mark"):
+            //fallthrough
```
👍

okkhoy commented on src/main/java/mona/input/Parser.java:
```diff
@@ -0,0 +1,217 @@
+        case ("mark"):
+            //fallthrough
+        case ("unmark"):
+            //fallthrough
```
shouldn't there be a break here?

okkhoy commented on src/main/java/mona/manager/TaskList.java:
```diff
@@ -0,0 +1,90 @@
+                    .collect(Collectors.toList());
+            ConsolePrint.printFilteredList(filteredTasks, keyword);
+        default:
+            //doNothing?
```
haha! "unknown command"?

okkhoy commented on src/main/java/mona/storage/Storage.java:
```diff
@@ -0,0 +1,236 @@
+            ArrayList<String> dataItems = readFile();
+            taskList = parse(dataItems);
+        } catch (IOException | MonaException e) {
+            e.printStackTrace();
```
In general, printing stack trace is considered bad exception handling

okkhoy commented on src/main/java/mona/storage/Storage.java:
```diff
@@ -0,0 +1,236 @@
+        try {
+            ArrayList<String> dataItems = readFile();
+            taskList = parse(dataItems);
+        } catch (IOException | MonaException e) {
```
better to catch 2 separate exceptions and show custom messages for each

okkhoy reviewed (commented):
In general, the code is clean. 💯 
 Please pay attention to spacing issues and intensity of comments within the code.

nur-haziq committed:
Update README.md

nur-haziq committed:
Update README.md

nur-haziq committed:
Update README.md

nur-haziq committed:
Update README.md

nur-haziq committed:
Update README.md

nur-haziq committed:
ConsolePrint.java: Refactor printFilteredList method

printFilteredList Each task printed now has an index that corresponds
to their index in the original list. This enables users to find the
index of their intended task to perform operations such as
mark/unmark/delete.

nur-haziq committed:
README.md: Add details to 'find' command

nur-haziq committed:
TaskList.java: Refactor code to allow for case insensitive search

'find' command will no longer be case sensitive

nur-haziq committed:
Add vertical spacing between methods to improve readability

nur-haziq committed:
Add custom error messages for incorrect user input

nur-haziq committed:
feat: Add ability for users to see a list of all available commands

Users can now input 'help' to see a list of all available commands and
how to use them

nur-haziq committed:
Parser.java: Refactor 'bye' command to ignore extra parameters

eg. an input of 'bye 123' will be treated as 'bye'

nur-haziq committed:
Add custom reply messages after each successful command

nur-haziq committed:
README.md: Add command types to the hyperlink section

nur-haziq committed:
Add error handling for some instances of corrupted data files

nur-haziq committed:
README.md: Add guide on how to edit the data file

nur-haziq committed:
README.md: Add section on possible issues faced during startup

nur-haziq committed:
README.md: Fix indentations for bullet points
